pkg/jobs/workers/mail.go

```diff
@@ -53,12 +58,61 @@ type MailPart struct {
 
 // SendMail is the sendmail worker function.
 func SendMail(ctx context.Context, m *jobs.Message) error {
-	var opts MailOptions
-	if err := m.Unmarshal(&opts); err != nil {
+	opts := &MailOptions{}
+	err := m.Unmarshal(&opts)
+	if err != nil {
 		return err
 	}
+	domain := ctx.Value(jobs.ContextDomainKey).(string)
+	switch opts.Mode {
+	case "noreply":
+		toAddr, err := addressFromDomain(domain)
+		if err != nil {
+			return err
+		}
+		opts.To = []*MailAddress{toAddr}
+		opts.From = &MailAddress{Email: "noreply@" + domain}
+	case "from":
+		fromAddr, err := addressFromDomain(domain)
+		if err != nil {
+			return err
+		}
+		opts.From = fromAddr
+	default:
+		return fmt.Errorf("Mail sent with unknown mode %s", opts.Mode)
+	}
+	return sendMail(ctx, opts)
+}
+
+func addressFromDomain(domain string) (*MailAddress, error) {
+	in, err := instance.Get(domain)
+	if err != nil {
+		return nil, err
+	}
+	doc := &couchdb.JSONDoc{}
+	err = couchdb.GetDoc(in, consts.Settings, consts.InstanceSettingsID, doc)
+	if err != nil {
+		return nil, err
+	}
+	email, ok := doc.M["email"].(string)
+	if !ok {
+		return nil, fmt.Errorf("Domain %s has no email in its settings", domain)
+	}
+	return &MailAddress{
+		Name:  "", // TODO: no name settings for an instance ?
```

Comment:
@jinroh https://github.com/cozy/cozy-stack/blob/master/docs/settings.md#get-settingsinstance

It's not yet used. It will be set by the onboarding app later.